File: Sprint Web/week 4/src/Entity/Matchs.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Matchs
 *
 * @ORM\Table(name="matchs", indexes={@ORM\Index(name="id_equipe1", columns={"id_equipe1"}), @ORM\Index(name="id_stade", columns={"id_stade"}), @ORM\Index(name="id_equipe2", columns={"id_equipe2"}), @ORM\Index(name="id_org", columns={"id_org"})})
 * @ORM\Entity(repositoryClass="App\Repository\MatchsRepository")
 */
class Matchs
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_match", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idMatch;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date", type="date", nullable=false)
     * @Assert\GreaterThanOrEqual("today")
     */
    private $date;

    /**
     * @var string
     *  @Assert\NotNull
     * @Assert\NotBlank
     * @ORM\Column(name="heure_deb", type="string", length=255, nullable=false)
     */
    private $heureDeb;

    /**
     * @var string
     * @Assert\NotNull
     * @ORM\Column(name="heure_fin", type="string", length=255, nullable=false)
     * @Assert\NotBlank
     * @Assert\Expression (
     *     "this.getHeureDeb() < this.getHeureFin()",
     *     message="l'heure de fin ne doit pas etre avant l'heure de début"
     * )
     */
    private $heureFin;

    /**
     * @var string
     * @Assert\NotNull
     * @ORM\Column(name="description", type="string", length=255, nullable=false)
     * @Assert\NotBlank
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="status", type="string", length=255, nullable=false)
     */
    private $status;

    /**
     * @var string|null
     *
     * @ORM\Column(name="score", type="string", length=255, nullable=true)
     */
    private $score;

    /**
     * @var int|null
     * @Assert\NotNull
     * @ORM\Column(name="nb_spectateur", type="integer", nullable=true)
     * @Assert\NotBlank
     */
    private $nbSpectateur;

    /**
     * @var \Equipe
     *
     * @ORM\ManyToOne(targetEntity="Equipe")
     * @Assert\Expression (
     *     "this.getIdEquipe1() != this.getIdEquipe2()",
     *     message=" equipe1 ne doit pas etre equipe2"
     * )
     *
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_equipe1", referencedColumnName="id_e")
     * })
     */
    private $idEquipe1;

    /**
     * @var \Equipe
     *
     * @ORM\ManyToOne(targetEntity="Equipe")
     * * @Assert\Expression (
     *     "this.getIdEquipe1() != this.getIdEquipe2()",
     *     message=" equipe1 ne doit pas etre equipe2"
     * )
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_equipe2", referencedColumnName="id_e")
     * })
     */
    private $idEquipe2;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_org", referencedColumnName="id")
     * })
     */
    private $idOrg;

    /**
     * @var \Stade
     *
     * @ORM\ManyToOne(targetEntity="Stade")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_stade", referencedColumnName="id_stade")
     * })
     */
    private $idStade;

    /**
     * Get Date
     *
     * @return DateTime
     */
    public function getDate()
    {
        return $this->date;
    }

    /**
     * Get heureDeb
     *
     * @return string
     */
    public function getHeureDeb()
    {
        return $this->heureDeb;
    }


    public function getHeureFin(): ?string
    {
        return $this->heureFin;
    }

    public function getNbSpectateur(): ?int
    {
        return $this->nbSpectateur;
    }
    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function getIdMatch(): ?int
    {
        return $this->idMatch;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }
    public function getScore(): ?string
    {
        return $this->score;
    }
    /**
     * Get IdEquipe1
     *
     * @return integer
     */
    public function getIdEquipe1()
    {
        return $this->idEquipe1;
    }
    /**
     * Get IdEquipe2
     *
     * @return integer
     */
    public function getIdEquipe2()
    {
        return $this->idEquipe2;
    }
    /**
     * Get IdOrg
     *
     * @return integer
     */
    public function getIdOrg()
    {
        return $this->idOrg;
    }
    /**
     * Get IdStade
     *
     * @return integer
     */
    public function getIdStade()
    {
        return $this->idStade;
    }


    public function setDate(\DateTime $date)
    {
        $this->date = $date;
        return $this;
    }

    /*   public function setIdMatch(int $idMatch)
       {
           $this->idMatch = $idMatch;
           return $this;
       }*/


    public function setHeureDeb(string $heureDeb)
    {
        $this->heureDeb = $heureDeb;
        return $this;
    }


    public function setHeureFin(string $heureFin)
    {
        $this->heureFin = $heureFin;
        return $this;
    }

    public function setNbSpectateur(int $nbSpectateur)
    {
        $this->nbSpectateur = $nbSpectateur;
        return $this;
    }

    public function setDescription(string $description)
    {
        $this->description = $description;
        return $this;
    }

    public function setStatus(string $st)
    {
        $this->status = $st;
        return $this;
    }

    public function setScore(string $score)
    {
        $this->score = $score;
        return $this;
    }

    /**
     * Set idEquipe1
     *
     * @param integer $idE
     *
     */
    public function setIdEquipe1($idE)
    {
        $this->idEquipe1 = $idE;
        return $this;
    }


    /**
     * Set idEquipe2
     *
     * @param integer $idE2
     *
     */
    public function setIdEquipe2($idE2)
    {
        $this->idEquipe2 = $idE2;
        return $this;
    }
    /**
     * Set idOrg
     *
     * @param integer $idOrg
     *
     */
    public function setIdOrg($idOrg)
    {
        $this->idOrg = $idOrg;
        return $this;
    }

    /**
     * Set idStade
     *
     * @param integer $idStade
     *
     */
    public function setIdStade($idStade)
    {
        $this->idStade = $idStade;
        return $this;
    }

}

File: SprintMobile/src/com/gestionMatch/entities/Matchs.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.gestionMatch.entities;

import java.util.Date;

/**
 *
 * @author esprit
 */
public class Matchs {
    private int idMatch ;
    private int idStade;
    private int idEquipe1;
    private int idEquipe2;
    private Date date;
    private int nbSpectateur;
    private String heureDeb;
    private String heureFin;
    private String description;
    private String status;
    private String score;
    private int idOrg;

    public Matchs() {
    }

    public Matchs(int idMatch, int idStade, int idEquipe1, int idEquipe2, Date date, int nbSpectateur, String heureDeb, String heureFin, String description, String status, String score, int idOrg) {
        this.idMatch = idMatch;
        this.idStade = idStade;
        this.idEquipe1 = idEquipe1;
        this.idEquipe2 = idEquipe2;
        this.date = date;
        this.nbSpectateur = nbSpectateur;
        this.heureDeb = heureDeb;
        this.heureFin = heureFin;
        this.description = description;
        this.status = status;
        this.score = score;
        this.idOrg = idOrg;
    }

    public Matchs(int idStade, int idEquipe1, int idEquipe2, Date date, int nbSpectateur, String heureDeb, String heureFin, String description, String status, String score, int idOrg) {
        this.idStade = idStade;
        this.idEquipe1 = idEquipe1;
        this.idEquipe2 = idEquipe2;
        this.date = date;
        this.nbSpectateur = nbSpectateur;
        this.heureDeb = heureDeb;
        this.heureFin = heureFin;
        this.description = description;
        this.status = status;
        this.score = score;
        this.idOrg = idOrg;
    }

    
    
    
    public int getIdMatch() {
        return idMatch;
    }

    public void setIdMatch(int idMatch) {
        this.idMatch = idMatch;
    }

    public int getIdStade() {
        return idStade;
    }

    public void setIdStade(int idStade) {
        this.idStade = idStade;
    }

    public int getIdEquipe1() {
        return idEquipe1;
    }

    public void setIdEquipe1(int idEquipe1) {
        this.idEquipe1 = idEquipe1;
    }

    public int getIdEquipe2() {
        return idEquipe2;
    }

    public void setIdEquipe2(int idEquipe2) {
        this.idEquipe2 = idEquipe2;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public int getNbSpectateur() {
        return nbSpectateur;
    }

    public void setNbSpectateur(int nbSpectateur) {
        this.nbSpectateur = nbSpectateur;
    }

    public String getHeureDeb() {
        return heureDeb;
    }

    public void setHeureDeb(String heureDeb) {
        this.heureDeb = heureDeb;
    }

    public String getHeureFin() {
        return heureFin;
    }

    public void setHeureFin(String heureFin) {
        this.heureFin = heureFin;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getScore() {
        return score;
    }

    public void setScore(String score) {
        this.score = score;
    }

    public int getIdOrg() {
        return idOrg;
    }

    public void setIdOrg(int idOrg) {
        this.idOrg = idOrg;
    }

    @Override
    public String toString() {
        return "Matchs{" + "idMatch=" + idMatch + ", idStade=" + idStade + ", idEquipe1=" + idEquipe1 + ", idEquipe2=" + idEquipe2 + ", date=" + date + ", nbSpectateur=" + nbSpectateur + ", heureDeb=" + heureDeb + ", heureFin=" + heureFin + ", description=" + description + ", status=" + status + ", score=" + score + ", idOrg=" + idOrg + '}';
    }
    
    
}
